src/ui/pages/Connections/Connections.tsx

```diff
@@ -49,313 +56,328 @@ import "./Connections.scss";
+      return showConnections;
+    }, [showConnections]);
 
-  const deleteConnection = async () => {
-    if (!deletePendingItem) return;
+    useSwipeBack(getConnectionsTab, canStart, () => setShowConnections(false));
 
-    try {
-      setDeletePendingItem(null);
-      await Agent.agent.connections.deleteStaleLocalConnectionById(
-        deletePendingItem.id
-      );
-      dispatch(setToastMsg(ToastMsgType.CONNECTION_DELETED));
-      dispatch(removeConnectionCache(deletePendingItem.id));
-    } catch (error) {
-      showError(
-        "Unable to delete connection",
-        error,
-        dispatch,
-        ToastMsgType.DELETE_CONNECTION_FAIL
+    const deletePendingCheckProps = useMemo(
+      () => ({
+        title: i18n.t("connections.tab.detelepending.title"),
+        description: i18n.t("connections.tab.detelepending.description"),
+        button: i18n.t("connections.tab.detelepending.button"),
+      }),
+      []
+    );
+
+    const deleteConnection = async () => {
+      if (!deletePendingItem) return;
+
+      try {
+        setDeletePendingItem(null);
+        await Agent.agent.connections.deleteStaleLocalConnectionById(
+          deletePendingItem.id
+        );
+        dispatch(setToastMsg(ToastMsgType.CONNECTION_DELETED));
+        dispatch(removeConnectionCache(deletePendingItem.id));
+      } catch (error) {
+        showError(
+          "Unable to delete connection",
+          error,
+          dispatch,
+          ToastMsgType.DELETE_CONNECTION_FAIL
+        );
+      }
+    };
+
+    const ConnectionsBody = () => {
+      return (
+        <div className="connections-tab-center">
+          <IonContent className="connections-container">
+            <IonGrid>
+              <IonRow>
+                <IonCol size="12">
+                  {mappedConnections.map((alphabeticGroup, index) => {
+                    return (
+                      <IonItemGroup
+                        className="connections-list"
+                        key={index}
+                      >
+                        <IonItemDivider id={alphabeticGroup.key}>
+                          <IonLabel>{alphabeticGroup.key}</IonLabel>
+                        </IonItemDivider>
+                        <AlphabeticList
+                          items={Array.from(alphabeticGroup.value)}
+                          handleShowConnectionDetails={
+                            handleShowConnectionDetails
+                          }
+                        />
+                      </IonItemGroup>
+                    );
+                  })}
+                </IonCol>
+              </IonRow>
+            </IonGrid>
+          </IonContent>
+          <AlphabetSelector />
+        </div>
       );
-    }
-  };
+    };
 
-  return (
-    <>
-      <SideSlider isOpen={showConnections}>
-        <TabLayout
-          hardwareBackButtonConfig={backHardwareConfig}
+    return (
+      <>
+        {selfPaginated ? (
```

Comment:
Because the menu has its own header and buttons, same with the original Connections component which was originally designed to render as a full tab, while now we only need to render the content inside the Menu/sub-menu frame.

Reply:
ok but what about when we have no connections and need the placeholder? does that work already?

Reply:
Forgot to add that to the PR, I need to add a small condition to Menu and pass down that in TabLayout in the same way as it's done in the original component. Sorry forgot about that too, refactoring this wasn't easy and it just slipped through the cracks.

Reply:
Actually there's an easier way, I'll just render that inside the exported component directly.